### DISCUSSION TITLE
touches array in touchstart event is undefined

### DISCUSSION THREAD

**Discussion by aeki-admin** [pixijs/pixijs#9197] [Q&A]
version:  pixi.js-legacy": "7.1.1"
```
const zone = this.stage.addChild(this.elements.zone) // new Graphics() 
zone.cursor = 'pointer'
zone.on('touchstart', (event) => {
      console.log(event.touches ||  event.data.pointerEvent.touches) // undefined 
})
```

How can I access touches in touch events? 

Thanks!

**Comment by bigtimebuddy** [comment]
Try `event.nativeEvent.touches`

http://pixijs.download/dev/docs/PIXI.FederatedEvent.html#nativeEvent

**Reply by aeki-admin** [reply]
Hello, still undefined. I'm on the ios simulator, would that be an issue?

**Reply by SuperSodaSea** [reply]
> <https://caniuse.com/?search=touches>

Seems that `touches` is not supported on Safari (but supported on Safari on iOS)?

Maybe you can try adding a `touchstart` listener to a native DOM element and see whether `e.touches` exists?

**Reply by SuperSodaSea** [reply]
OK now I know the reason: We only listen `pointerdown` and dispatch `touchstart` if `e.pointerType === 'touch'`, so there is no `touches` in the `nativeEvent` (which is `PointerEvent` instead of `TouchEvent`).

**Reply by aeki-admin** [reply]
Thanks for replying, `e.touches` exists on the DOM so is that the only way to get `touches` for now?

**Reply by SuperSodaSea** [reply]
Yeah, if you need this feature for objects in PixiJS, you can make a [Feature Request](https://github.com/pixijs/pixijs/issues/new?assignees=&labels=Feature+Request&template=feature_request.yml&title=Feature+Request%3A+).
